airbyte-integrations/connectors/source-snowflake/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/SnowflakeJdbcSourceAcceptanceTest.java

```diff
@@ -96,9 +98,34 @@ public AbstractJdbcSource<JDBCType> getJdbcSource() {
 
   @Test
   void testCheckFailure() throws Exception {
-    ((ObjectNode) config.get("credentials")).put("password", "fake");
+    ((ObjectNode) config).with("credentials").put(JdbcUtils.PASSWORD_KEY, "fake");
     final AirbyteConnectionStatus actual = source.check(config);
     assertEquals(Status.FAILED, actual.getStatus());
+    assertTrue(actual.getMessage().contains("State code: 08001; Error code: 390100;"));
+  }
+
+  @Test
+  public void testCheckIncorrectUsernameFailure() throws Exception {
+    ((ObjectNode) config).with("credentials").put(JdbcUtils.USERNAME_KEY, "fake");
+    final AirbyteConnectionStatus actual = source.check(config);
+    assertEquals(Status.FAILED, actual.getStatus());
+    assertTrue(actual.getMessage().contains("State code: 08001; Error code: 390100;"));
+  }
+
+  @Test
+  public void testCheckEmptyUsernameFailure() throws Exception {
+    ((ObjectNode) config).with("credentials").put(JdbcUtils.USERNAME_KEY, "");
+    final AirbyteConnectionStatus actual = source.check(config);
+    assertEquals(Status.FAILED, actual.getStatus());
+    assertTrue(actual.getMessage().contains("State code: 28000; Error code: 200011;"));
+  }
+
+  @Test
+  public void testCheckIncorrectHostFailure() throws Exception {
+    ((ObjectNode) config).put(JdbcUtils.HOST_KEY, "localhost2");
+    final AirbyteConnectionStatus actual = source.check(config);
+    assertEquals(Status.FAILED, actual.getStatus());
+    assertTrue(actual.getMessage().contains("State code: 28000; Error code: 200028;"));
   }
 
   @Override
```

Comment:
As a side comment, @cgardens has a [draft PR](https://github.com/airbytehq/airbyte/pull/15866/files) that covers some tests not covered here, namely invalid schema and database. Let's try to keep parity with tests since it appears that Postgres and other connectors already cover some of these tests

Also, can you link where the error codes were found? It help to be able to verify where `200028`, `200011`, etc are mentioned in their own documentation. For instance, `390100` is not found within their documentation for [Federated Authentication & SSO Error Codes](https://docs.snowflake.com/en/user-guide/errors-saml.html) so curious how you found these codes?

Reply:
@ryankfu as for the tests - I don't think our tests will contradict each other. In some cases, they may cover the same case on different datasets. Therefore, I do not think that it can interfere.

Reply:
@ryankfu as for the codes - now I don't process them, but only take the state code either from the **exception** or from **cause of this exception**. This code that I output is returned to me in a `SQLException`.

Reply:
It seems the messaging was lost, the comment was not intended to say that Charles has a PR that would potentially conflict with the changes here. The message was that here are the list of tests that should be covered in each of the different connectors. So namely invalid:
* host
* port
* username
* password
* database
* schema

The reason for asking for how the error/state codes were found is because some of the mongodb tests are asserting something like 
```
assertTrue(airbyteConnectionStatus.getMessage().contains("State code: -4"));
```
However from the standpoint of debugging, if someone wanted to look up what mongo db state code is how can it be determined that they're the right state codes? For instance looking up state codes for mongodb I could only find [these state code](https://www.mongodb.com/docs/v6.0/reference/exit-codes/) but none of them match `-4`. It begs the question, how would these state codes be useful if documentation from mongodb doesn't match the codes in the test and what they mean?

Reply:
@ryankfu yes, I agree with you that in some cases posting the code may not give results, but in most cases it is useful information that really helps the user. I can't tell you exactly why the driver developers do this (this is about mongodb), or why not all state codes are in the documentation (this is about your snowflake comment), but for most connectors all state codes are in the documentation and up a normal description is attached to them. This code is intended for general processing - and it handles messages from different connectors in the same way. Unfortunately, I cannot guarantee the accuracy of the state code, or the accuracy of the exception message.